File: 0567-permutation-in-string/0567-permutation-in-string.cpp
class Solution {
public:
    bool checkInclusion(string s1, string s2) {
        int arr[26]={0};
        int brr[26]={0};
        for(char c:s1){
            arr[c-'a']++;
        }
        int left=0, right=0;
        while(right < s2.length()){
            brr[s2[right]-'a']++;
            
            if(right-left+1==s1.length()){
                if (equal(begin(arr), end(arr), begin(brr))) {
                    return true;
                }
                brr[s2[left]-'a']--;
                left++;
            }
            right++;
        }
        return false;

    }
};

File: 0567-permutation-in-string/0567-permutation-in-string.java
class Solution {
    public boolean checkInclusion(String s1, String s2) {
        int[] s1Freq = new int[26];
        int[] s2Freq = new int[26];
        int left = 0;
        int n = s1.length(), m = s2.length();
        for(char c: s1.toCharArray()) s1Freq[c-'a']++;
        for(int right = 0;right <m ;right++){
            s2Freq[s2.charAt(right)-'a']++;
            if(right-left+1 > n){
                s2Freq[s2.charAt(left)-'a']--;
                left++;
            }
            if(right-left+1 == n){
                int flag = 0;
                for(int i=0;i<26;i++){
                    if(s1Freq[i]!=s2Freq[i]) 
                    {
                        flag = 1;
                        break;
                    }
                }
                if(flag==0) return true;

            }

        }
        return false;
    }
}